libraries/base/java-utils/Utils.java

```diff
@@ -123,6 +124,34 @@ public static String byteBufferToStr(ByteBuffer buffer)
         return new String(bytes, "UTF-8");
     }
 
+    public static ByteBuffer c_memcpy(ByteBuffer dest, ByteBuffer src, int size) {
+        ByteBuffer src2 = src.duplicate();
+        ByteBuffer dest2 = dest.duplicate();
+        MemoryManager.bufSetLimit(src2, size);
+        dest2.put(src2);
+        return dest2;
```

Comment:
Apologize - my last claim about returning `dest2` was wrong! I thought about it some more and realised that the returned buffer *must* point to the start of `dest` while `dest2` will point to after the newly written data. So return `dest`. Do the same for `memmove` as well and it should be good.